### DISCUSSION TITLE
Generic data access object

### DISCUSSION THREAD

**Discussion by programmervarunshridhar** [simolus3/drift#992] [Q&A] [answered]
Hi,

Is it possible to write a generic data access object?

For example

Instead of writing

`select(customers).get();`

I want to write

`select<Customer>.get()`

and therefore, I can have a generic DAO class as below

`abstract class DAO<TMoorModel extends Table>{
   Future<Iterable<TMoorModel>> getAll(){
     return select<TMoorModel>.get();
  }
}`

I wouldn't have to implement the getAll method in my concrete DAOs

A concrete DAO would look as below

`class CustomerDAO extends DAO<Customer>{}`

Is there a generic implementation?

Thanks in advance.

**Comment by simolus3** [comment, marked as answer]
Yes, you can write some generic daos with moor. You can check [this issue](https://github.com/simolus3/moor/issues/333#issuecomment-664369197) for an example.
